openqa-label-known-issues

```diff
@@ -51,18 +52,19 @@ handle_unreachable() {
         return 1
     fi
     # resorting to downloading the job details page instead of the
-    # log, overwrites $out
-    if ! curl "${curl_args[@]}" -s "$testurl" -o "$out"; then
+    # log
+    if ! curl "${curl_args[@]}" -s "$testurl" -o "${html_out}"; then
         echoerr "'$testurl' can be reached but not downloaded, bailing out"
         curl "${curl_args[@]}" "$testurl"
         exit 2
     fi
 
-    if hxnormalize -x "$out" | hxselect -s '\n' -c '.links_a .resborder' | grep -qPzo '(?s)Gru job failed.*connection error.*Inactivity timeout'; then
+    if hxnormalize -x "${html_out}" | hxselect -s '\n' -c '.links_a .resborder' | grep -qPzo '(?s)Gru job failed.*connection error.*Inactivity timeout'; then
         "${client_call[@]}" -X POST jobs/"$id"/comments text='poo#62456 test incompletes after failing in GRU download task on "Inactivity timeout" with no logs'
         "${client_call[@]}" -X POST jobs/"$id"/restart
         return 1
     fi
+    timeago=$(grep timeago "$html_out" | hxselect -s '\n' -c '.timeago::attr(title)')
```

Comment:
global, undeclared variables should be avoided.
Would it make sense to just move this call back into `handle_unreachable`?
What was the reason to extract it in the first place?

Reply:
I am not if this better, but at least I declared the variable. wdyt?

Reply:
I can tell you from > 25 years of experience that it might seem convenient to just make a variable global, and yes in bash that's even more often the case, but it creates technical debt.
It's hard to avoid for the output files that we have to remove in the exit handler, but don't just do it because it seems convenient.
In my commit I moved that statement back into `_handle_unreachable`.